docs/explainer.md

```diff
@@ -160,4 +162,45 @@ let writableStream = tcpSocket.writable;
 tcpSocket.close();
 ```
 
+To simplify security analysis, an API for listening for incoming connections is not yet being proposed.
+
 ## UDP
+
+To simplify security analysis, the initial proposal only supports cases where the web app initiates communication with a remote host.
+
+```
+const options = {
+    remoteAddress: 'example.com',
+    remotePort: 7
+};
+
+try {
+  const udpSocket = await navigator.openUDPSocket(options);
+  doStuffWith(udpSocket);
+  ...
+} catch (err) {
+  // handle error
+} finally {
+  udpSocket.close();
+}
+```
+
+The `remoteAddress` member may be omitted or ignored - the user agent may invite the user to specify the address.
+
+There is currently no provision for setting the local address or port.
+
+`send` returns a promise, that resolves when a message is sent, or the send fails:
+
+```
+let blob = ...;
+await udpSocket.send(blob, {timeout: 5000});
```

Comment:
I thought you didn't want a timeout since it would imply an extra level of buffering in the user agent before it gets to the kernel?

Reply:
Also were we going to have an analogue for `select` that lets you wait until the port is available? Or would that be automatically done by calling `send`?

Reply:
You are right. I have removed the timeout. The `select` might end up complicating implementation. We can circle back if/when needed.